### PR TITLE
Add side menu component

### PR DESCRIPTION
**Pull request by CyberSlug322**

### CODE REVIEW

File: src/containers/sideMenu/SideMenuBlock.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+.side-menu-block {
```
Comment (anton-matseichyk) [inline src/containers/sideMenu/SideMenuBlock.scss:2]: I don't think we need this interim component. These styles are positioning, so they belong to .layout__side-menu

File: src/containers/sideMenu/SideMenuBlock.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { SideMenu } from "./SideMenu.jsx";
+import './SideMenuBlock.scss'
+
+export const SideMenuBlock = ({onMenuButtonClick}) => (
```
Comment (anton-matseichyk) [inline src/containers/sideMenu/SideMenuBlock.jsx:5]: onOutsideClick

Comment (anton-matseichyk) [reply src/containers/sideMenu/SideMenuBlock.jsx:5]: Check event.target and event.relatedTarget(?) to figure out whether div itself or child was clicked

**Review by anton-matseichyk — CHANGES_REQUESTED** [review]

File: src/containers/sideMenu/SideMenuBlock.scss (hunk 1 above)
Comment (CyberSlug322) [reply src/containers/sideMenu/SideMenuBlock.scss:2]: should i the whole component SideMenuBlock or move css styles to layout is enought?

File: src/containers/sideMenu/SideMenuBlock.jsx (hunk 2 above)
Comment (CyberSlug322) [reply src/containers/sideMenu/SideMenuBlock.jsx:5]: i used stopPropagation() is it fine?

File: src/containers/sideMenu/SideMenuBlock.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    position: absolute;
```
Comment (anton-matseichyk) [inline src/containers/sideMenu/SideMenuBlock.scss:2]: Still not resolved. These styles need 
a) bugfix. Try loading a long list of beers, scrolling to the bottom of the page and click outside. 
b) moving to layout__side-menu. These are positioning styles. It is parent's responsibility to provide a layer for SideMenuBlock. SideMenuBlock should just render 100% of width and height of this layer.

File: src/containers/sideMenu/SideMenuBlock.jsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { SideMenu } from "./SideMenu.jsx";
+import './SideMenuBlock.scss';
+
+export const SideMenuBlock = ({onMenuButtonClick, className}) => (
+
+    <div onClick={onMenuButtonClick} className="side-menu-block">
+        <SideMenu className={className}/>
```
Comment (anton-matseichyk) [inline src/containers/sideMenu/SideMenuBlock.jsx:8]: BEM elements work only for immediate children, not drilling to grandchildren

File: src/containers/sideMenu/SideMenu.jsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { SideMenuHeader } from "./SideMenuHeader.jsx";
 import { SideMenuNavigation } from "./SideMenuNavigation.jsx";
 
 export const SideMenu = ({className}) => (
-  <div className={`side-menu ${className}`}> 
+  <div className={`side-menu ${className}`} onClick={(e)=> e.stopPropagation()}>
```
Comment (anton-matseichyk) [inline src/containers/sideMenu/SideMenu.jsx:7]: Will discuss that tomorrow

**Review by anton-matseichyk — CHANGES_REQUESTED** [review]

**Pull request merged**